### PR TITLE
Add get-topics endpoint

### PR DESCRIPTION
**Pull request by zacwong2151**
1. Add endpoint to GET question topics
2. Add error handling for duplicate question title when doing a POST/PUT request

### CODE REVIEW

**Review by wjayee — APPROVED** [review]
LGTM!

File: question-service/src/routes/questionsRoute.ts [hunk 1]
```diff
@@ -61,8 +79,13 @@ router.post("/", async (req: Request, res: Response): Promise<Response> => {
         return res.status(200).send({
             message: "Question successfully created",
         });
-    } catch (error) {
+    } catch (error: any) {
         console.log(error);
+        if (error.code === 11000) {
```
Comment (brein62) [inline question-service/src/routes/questionsRoute.ts:84]: would be nice to identify what this error code is. as far as i can see this looks like a magic number

**Review by brein62 — APPROVED** [review]
LGTM! just one minor thing

File: question-service/src/routes/questionsRoute.ts (hunk 1 above)
Comment (zacwong2151) [reply question-service/src/routes/questionsRoute.ts]: gd point thanks! fixed

**Pull request merged**